python-api/tests/client_test.py

```diff
@@ -0,0 +1,224 @@
+#
+# Licensed to Cloudera, Inc. under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  Cloudera, Inc. licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import pytest
+import responses
+import socket
+from livy.client import HttpClient
+from urlparse import urlparse
+from ConfigParser import NoOptionError
+import pathlib2
+import threading
+import os
+
+session_id = 0
+job_id = 1
+base_uri = 'http://{}:{}'.format(socket.gethostname(), 8998)
+client_test = None
+invoked_queued_callback = False
+invoked_running_callback = False
+
+
+@responses.activate
+def mock_and_validate_create_new_session(defaults):
+    global client_test
+    create_session_request_mock_uri = base_uri + "/sessions/"
+    app_name = 'Test Client App'
+    conf_dict = {'spark.app.name': app_name}
+    responses.add(responses.POST, create_session_request_mock_uri,
+                  json={u'kind': u'pyspark', u'log': [], u'proxyUser': None,
```

Comment:
I kinda dislike this kind of alignment, which you're using everywhere. If for some reason you need to change the first line of the statement, now you have to change every single line to fix the alignment.

Just used indentation like we do for Java/Scala.